### PR TITLE
[BACKLOG-9928] Improve XMI generation to use DatabaseMetadata

### PR DESCRIPTION
**Pull request by e-cuellar**

### CODE REVIEW

**Comment by e-cuellar** [comment]
@mdamour1976 Please Sir!

File: core/src/org/pentaho/di/core/database/Database.java [hunk 1]
```diff
@@ -2173,6 +2157,73 @@ public RowMetaInterface getQueryFields( String sql, boolean param, RowMetaInterf
...
+        try {
+          preparedStatement.close();
+        } catch ( SQLException e ) {
+          throw new KettleDatabaseException( "Unable to close prepared statement after determining SQL layout", e );
+        }
+      }
+    }
+  }
+
+  private RowMetaInterface getQueryFieldsFromDatabaseMetaData( String sql, boolean param, RowMetaInterface inform,
```
Comment (mdamour1976) [inline core/src/org/pentaho/di/core/database/Database.java]: @e-cuellar just being extra picky, but how many of these parameters are actually used in this method?  Can we chop it down to only the ones used?

Comment (e-cuellar) [inline core/src/org/pentaho/di/core/database/Database.java]: Fixed!

**Comment by mdamour1976** [comment]
@e-cuellar let's work today to add some unit tests (which probably should have existing long ago) to DatabaseTest.

**Pull request merged**